### DISCUSSION TITLE
Using App router

### DISCUSSION THREAD

**Discussion by rhtmhr895** [vercel/next.js#50145] [Help]
### Summary

I am using the new App router feature of NEXT.JS and building an app, I am using a new way for data fetching i.e. :
async function getData() {
  const res = await fetch('https://api.example.com/...');
  // The return value is *not* serialized
  // You can return Date, Map, Set, etc.
 
  // Recommendation: handle errors
  if (!res.ok) {
    // This will activate the closest `error.js` Error Boundary
    throw new Error('Failed to fetch data');
  }
 
  return res.json();
}:

does this enables server side rendring in my application and If yes then why the build  that is being generated is rendered as static ---->(Static)  automatically rendered as static HTML (uses no initial props) and not getting rendered on server side.
can anyone help me in this ?

**Comment by icyJoseph** [comment]
By default Next.js pre-renders pages, and generates an actual HTML page. If you want to change this default behaviour, you need to work with the `fetch` cache options. 

- https://nextjs.org/docs/app/building-your-application/data-fetching/fetching#static-data-fetching

In your case, you could use:

```ts
fetch('https://...', { cache: 'no-store' });
```

Note that there's a client side, per RSC page cache:

- https://nextjs.org/docs/app/building-your-application/routing/linking-and-navigating#client-side-caching-of-rendered-server-components

Maybe you'll find this article interesting: https://www.inngest.com/blog/5-lessons-learned-from-taking-next-js-app-router-to-production

**Reply by rhtmhr895** [reply]
does using "use server" allows me to serve the app as server-side render instead of static rendering?

**Reply by icyJoseph** [reply]
I think you need to really go through the https://nextjs.org/docs/getting-started/react-essentials page, it'll cover a whole lot of your concerns.

The `use client` directive, is slightly confusing I'd say because, it is open ended, kind of by the spec. For example, many tech influencers like to say that these components run only on the client, but that's not necessarily 100% true. 

In Next.js, pre-rendering is seen as an advantage, because it leads to extremely low Time To Visible (although Time To Interactive might not have improved as much in the old pages directory). And it is also SEO friendly!

> With Server Components it possible to get low Time To Visible, and low Time To Interactive. 

So what Next.js does, is that it also pre-renders server side, those components marked as `use client`, https://nextjs.org/docs/getting-started/react-essentials#the-use-client-directive:

> Good to know: In Next.js, during the initial page load, both the rendered result of Server Components from the above step and Client Components are [pre-rendered on the server as HTML](https://nextjs.org/docs/app/building-your-application/rendering#static-and-dynamic-rendering-on-the-server) to produce a faster initial page load.

> Components in the Client Component module graph are primarily rendered on the client, but with Next.js, they can also be pre-rendered on the server and hydrated on the client.

At the moment, the only guarantee we have, is that everything that's not a client component (including those who are not using the `use client` directive, but are imported by one), are server components, and those run only server side.

> "use client" does not need to be defined in every file. The Client module boundary only needs to be defined once, at the "entry point", for all modules imported into it to be considered a Client Component.

I think what you want to do is "turn off" pre-rendering and, instead render on demand, on the Server exclusively. Right? You need to read about dynamic rendering. 

- https://nextjs.org/docs/app/building-your-application/rendering/static-and-dynamic-rendering#dynamic-rendering

And perhaps how to force dynamic rendering at segment level: https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config - most likely you want to start with `export const dynamic = "force-dynamic"`

But even that has to make sure that all components are server components, by not using stateful/interactive features: https://nextjs.org/docs/getting-started/react-essentials#when-to-use-server-and-client-components
